docs/syntax_and_semantics/offsetof.md

```diff
@@ -1,9 +1,19 @@
 # offsetof
 
-An `offsetof` expression returns the byte offset of an instance variable in a struct or class type. It accepts the type as first argument and the instance variable name prefixed by an `@` as second argument:
+An `offsetof` expression returns the byte offset of a subobject in an instance of a class or struct.
+
+There are two forms of `offsetof` expressions. The first form accepts any type as first argument and an instance variable name prefixed by an `@` as second argument:
+
+```crystal
+# x : Type
+offsetof(Type, @ivar) # the byte offset of x.@ivar relative to x
```

Comment:
Maybe the examples could be a tiny bit larger and self-contained?

I would move the explanation that `offsetof` returns the byte offset of x.@ivar relative to x (with x : Type) to prose text outside the example.
Dito below.